Signal/src/ViewControllers/HomeView/Chat List/ChatListViewController+Multiselect.swift

```diff
@@ -181,8 +181,7 @@ extension ChatListViewController {
         v.frame = CGRect(x: 0, y: 0, width: size.width, height: size.height)
         v.delegate = self
 
-        viewState.multiSelectState.contextMenuView = ContextMenuActionsViewContainer(v)
-        viewState.multiSelectState.contextMenuView?.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(hideMenu)))
+        viewState.multiSelectState.contextMenuView = ContextMenuActionsViewContainer(v, dismissCallback: hideMenu)
```

Comment:
I think you'll be creating a retain cycle here where hideMenu has a strong reference to the ChatListViewController instance which has a strong reference to its viewState which has a strong reference to contextMenuView, which now has a strong reference to hideMenu.  You should be able to fix it by making the callback' self reference weak (but then you'll want to pass an explicit block like:
dismissCallback: { [weak self] in self?.hideMenu() }

Reply:
Thank you for reviewing this PR, @harry-signal. Without changing the code, I added a `deinit` block inside `ContextMenuActionsViewContainer` and set a breakpoint inside of it. When I dismiss the menu, it hits the breakpoint. I believe that happens because of this [line](https://github.com/signalapp/Signal-iOS/blob/main/Signal/src/ViewControllers/HomeView/Chat%20List/ChatListViewController%2BMultiselect.swift#L214). Setting the `contextMenuView` to `nil` breaks the retain cycle. Please let me know if you'd still like me to make the suggested change.

Reply:
Thanks for the detailed reply (and sorry for my own delay). Yeah, you're right that removing the menu view on dismiss resolves the retain cycle. Still, it feels dangerous to have a temporary retain cycle like this, because its easy to make a future change that forgets to remove the circular reference and leaves things in a bad state. Its much safer to make the reference weak and never create a cycle in the first place. Of course in practice the weak reference isn't going to nil out, so it should be an easy change. Does that all make sense?

Reply:
I've made the change you requested. Please take a look and let me know if you have any other concerns.

Reply:
Thank you so much! I'll get this merged in and close this PR out, should be out in the next beta!

Reply:
Thank you. I noticed that you started the CI run and there's a linter failure. The failed script is called `Scripts/precommit.py` but it did not trigger a linter check for me prior to committing. Am I missing something in my environment setup?

Reply:
Yeah you can run the precommit script manually, or add it to your git configs precommit so it auto-lints things for you. Regardless, the way it works is I'll make a copy of this PR to merge into our internal repo, and in ~1 week the private repo will get pushed here to the public one and I will link this PR to the final commit and close it. I can run lint checks when I do that (they'll just run automatically) so no need to do anything on your end!

Reply:
@harry-signal This PR is still open. Wanted to follow up with you to see if everything went okay or if there's something I can do on my end to help resolve.

Reply:
Oh my bad, I merged this already in this PR: https://github.com/signalapp/Signal-iOS-Private/pull/4713. Github has weird UI

![image](https://user-images.githubusercontent.com/109690906/187312785-7b8cfc39-2259-4cf3-898b-b50ff4276bf1.png)

I'll explicitly mark this one closed. Thanks for sending this!

Reply:
Sweet! Thank you for the update.